src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -257,13 +255,14 @@ private static string GetIdentifierFromPath(string filePath)
             return builder.ToString();
         }
 
-        private static ParallelOptions GetParallelOptions()
+        private static ParallelOptions GetParallelOptions(GeneratorExecutionContext generatorExecutionContext)
         {
             if (Debugger.IsAttached)
             {
                 return new ParallelOptions { MaxDegreeOfParallelism = 1 };
             }
```

Comment:
Seeing this reminded me of another change we should make. The compiler parallelism can be explicitly disabled by users. This is a flag that generators should respect. Essentially if the user demands the compiler run in single threaded mode then generators loaded in the compiler should respect that. 

In the case that [CompilationOptions.ConcurrentBuild](http://sourceroslyn.io/#Microsoft.CodeAnalysis/Compilation/CompilationOptions.cs,ae09d3a82430aa43,references) is disabled I think we should have a `MaxDegreeOfParallelism` of one here or better avoid the `Parallel.For` entirely. 

Can take this up in a separate change though, just thought about it while reading this.

Reply:
Good tip -- I can cook up a follow-up PR to resolve this.